pkgs/development/python-modules/django/3.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, substituteAll
+, geos
+, gdal
+, asgiref
+, pytz
+, sqlparse
+, pythonOlder
+, withGdal ? false
+}:
+
+buildPythonPackage rec {
+  pname = "Django";
+  version = "3.1";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "2d390268a13c655c97e0e2ede9d117007996db692c1bb93eabebd4fb7ea7012b";
+  };
+
+  patches = stdenv.lib.optional withGdal
+    (substituteAll {
+      src = ./1.10-gis-libs.template.patch;
```

Comment:
This patch is not valid for django-3.

You can re-use the one I prepared when I did a django-3.0 PR (#92848)

```patch
diff -Nur a/django/contrib/gis/gdal/libgdal.py b/django/contrib/gis/gdal/libgdal.py
--- a/django/contrib/gis/gdal/libgdal.py	2020-07-09 22:34:05.330568948 +0100
+++ b/django/contrib/gis/gdal/libgdal.py	2020-07-09 22:35:08.679095615 +0100
@@ -14,7 +14,7 @@
     from django.conf import settings
     lib_path = settings.GDAL_LIBRARY_PATH
 except (AttributeError, ImportError, ImproperlyConfigured, OSError):
-    lib_path = None
+    lib_path = "@gdal@/lib/libgdal@extension@"

 if lib_path:
     lib_names = None
diff -Nur a/django/contrib/gis/geos/libgeos.py b/django/contrib/gis/geos/libgeos.py
--- a/django/contrib/gis/geos/libgeos.py	2020-07-09 22:34:05.331568941 +0100
+++ b/django/contrib/gis/geos/libgeos.py	2020-07-09 22:36:24.863526276 +0100
@@ -24,7 +24,7 @@
         from django.conf import settings
         lib_path = settings.GEOS_LIBRARY_PATH
     except (AttributeError, ImportError, ImproperlyConfigured, OSError):
-        lib_path = None
+        lib_path = "@geos@/lib/libgeos_c@extension@"

     # Setting the appropriate names for the GEOS-C library.
     if lib_path:
```